network_simulator/packet.py

```diff
@@ -0,0 +1,76 @@
+"""
+This module will hold the class Packet, subclasses of Packet, and various Packet-related Events.
+Packet will be an abstract class.
+"""
+
+from abc import ABCMeta, abstractmethod
+from .common import *
+
+
+class Packet(object):
+    """
+    Packets are units of information in the network.
+    They are generated by a flow; all packets in the same flow have the same destination network address.
+    Packets can be router packets, data packets, or ACKs. All of these have a fixed size specified in common.py.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, id=None, flow_ID=None, source_ID=None, destination_ID=None,
+                 start_time_sec=None, size_bits=None):
+        """
+        :ivar int id: an id field (int), used by the Flow to keep track of the packets and their ACKs,
+        and to know when to retransmit. These are zero-indexed.
+        :ivar int flow_ID: the flow to which this packet belongs. So, (flow_ID, packet_ID)
+        uniquely identifies a given data packet.
+        :ivar string source_ID: source Device ID (string).
+        :ivar string destination_ID: destination Device ID (string).
+        :ivar float start_time_sec: (float) time at which the packet was created (in sec).
+        :ivar int size_bits: (int) size of the packet in bits.
+        """
+        self.id = id
+        self.flow_ID = flow_ID
+        self.source_ID = source_ID
+        self.destination_ID = destination_ID
+        self.start_time_sec = start_time_sec
+        self.size_bits = size_bits
+
+
+class DataPacket(Packet):
+    """
+    DataPacket class: a packet used for transferring data between Hosts.
+    """
+
+    def __init__(self):
+        Packet.__init__(self)
+
+
+class AckPacket(Packet):
+    """
+    AckPacket class: a packet used for sending ACKs between Hosts.
+    """
+
+    def __init__(self, flow_packets_received=None):
+        Packet.__init__(self)
+
+        """
```

Comment:
The doc comment (between """) should be the first line in the function, before the __init__ call to the superclass.